deviceid/deviceid_nonwindows.go

```diff
@@ -14,33 +14,49 @@ import (
 )
 
 // Get returns a unique identifier for this device. The identifier is a random UUID that's stored on
-// disk at $HOME/.lanternsecrets/.deviceid. If unable to read/write to that location, this defaults to the
+// disk at `DataPath/deviceid`. If unable to read/write to that location, this defaults to the
 // old-style device ID derived from MAC address.
-func Get() string {
-	path := filepath.Join(appdir.InHomeDir(".lanternsecrets"))
-	err := os.Mkdir(path, 0755)
-	if err != nil && !os.IsExist(err) {
-		pterm.Error.Printfln("Unable to create folder to store deviceID, defaulting to old-style device ID, error: %s", err.Error())
-		return OldStyleDeviceID()
-	}
-
-	filename := filepath.Join(path, ".deviceid")
+func Get(dataPath string) string {
+	filename := filepath.Join(dataPath, "deviceid")
 	existing, err := os.ReadFile(filename)
 	if err != nil {
-		pterm.Debug.Println("Storing new deviceID")
-		_deviceID, err := uuid.NewRandom()
-		if err != nil {
-			pterm.Error.Printfln("Error generating new deviceID, defaulting to old-style device ID, error: %s", err.Error())
-			return OldStyleDeviceID()
+		deviceID := readFromHomeDir()
+		if deviceID == "" {
+			deviceID = newDeviceId()
 		}
```

Comment:
```suggestion
		deviceID := readFromHomeDir()
```
and move:
```
		if deviceID == "" {
			deviceID = newDeviceId()
}
```
inside readFromHomeDir

Reply:
I politely disagree. In my opinion a function should do one thing described in its name. Generating a new device ID inside the `readFromHomeDir` function would be confusing for a future reader.

However if you feel this is the better approach, I will implement it.